tree.go

```diff
@@ -415,6 +416,9 @@ walk: // Outer loop for walking the tree
 						p[i].Value = val
 					}
 
+					// replace p.value with p.key (pattern :)
+					relativePath = strings.Replace(relativePath, p[i].Value, ":"+p[i].Key, 1)
+
```

Comment:
Instead of calculating the `relativePath` on every request, you can save the [fullPath](https://github.com/youngbloood/gin/blob/adbfb90327a9e219af8a3543da29f9f71bd8f5ee/tree.go#L126) in the `node` `method: addRoute` so when you have the node you have also the `relativePath`. This means you have to add a field in the `node srtuct`.